File: dotNet-sdk/Semantria/Proxy/CategoryUpdateProxy.cs
using System;
using System.Collections.Generic;
using System.Text;
using Semantria.Com.Mapping.Configuration;
using Semantria.Com.Mapping.Configuration.Stub;

namespace Semantria.Com.Proxy
{
    public sealed class CategoryUpdateProxy : IUpdateProxy<Category>
    {
        #region Constructor

        private CategoryUpdateProxy()
        {
        }

        internal static IUpdateProxy<Category> CreateInstance()
        {
            return new CategoryUpdateProxy();
        }

        #endregion Constructor

        #region Properties

        private static string REMOVE_ALL_ITEMS_MARK = "*";
        private CategoryManagable _managable = new CategoryManagable();
        internal CategoryManagable Managable
        {
            get { return _managable; }
            set { _managable = value; }
        }

        #endregion Properties

        #region IUpdateProxy<Category> Members

        object IUpdateProxy<Category>.Stub
        {
            get { return _managable; }
        }

        void IUpdateProxy<Category>.Add(Category obj)
        {
            _managable.Added.Add(obj);
        }

        void IUpdateProxy<Category>.Remove(Category obj)
        {
            _managable.Removed.Add(obj.Name);
        }

        void IUpdateProxy<Category>.Edit(Category obj)
        {
            _managable.Added.Add(obj);
        }

        void IUpdateProxy<Category>.Clone(Category obj) {}

        void IUpdateProxy<Category>.RemoveAll()
        {
            _managable.Added.Clear();
            _managable.Removed.Clear();
            _managable.Removed.Add(REMOVE_ALL_ITEMS_MARK);
        }

        #endregion
    }
}

File: java-sdk/com/semantria/proxies/CategoryUpdateProxy.java
package com.semantria.proxies;

import com.semantria.interfaces.IUpdateProxy;
import com.semantria.objects.configuration.Category;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(name="categories")
public final class CategoryUpdateProxy implements IUpdateProxy<Category> 
{
	private List<Category> added;
	private List<String> removed;
	
	public CategoryUpdateProxy()
	{
		added = new ArrayList<Category>();
		removed = new ArrayList<String>();
	}
	
	public void add(Category cat)
	{
		added.add(cat);
	}
	
	public void remove(Category id)
	{
		removed.add(id.getName());
	}
	
	public void update(Category cat)
	{
		added.add(cat);
	}

	public void clone(Category item) {}

    public void removeAll() {
        added.clear();
        removed.clear();
        removed.add(REMOVE_ALL_ITEMS_MARK);
    }
	
	@XmlElementWrapper(name="added")
	@XmlElement(name="category")
	public List<Category> getAdded() { return added; }
	@XmlElementWrapper(name="removed")
	@XmlElement(name="category")
	public List<String> getRemoved() { return removed; }
	
}
